test/e2e/operator_test.go

```diff
@@ -250,79 +250,40 @@ func TestDNSLogging(t *testing.T) {
-	)
-	for _, ver := range co.Status.Versions {
-		if ver.Name == statuscontroller.OpenshiftCLIVersionName {
-			if len(ver.Version) == 0 {
-				break
+	for _, corednspod := range coreDNSPods.Items {
+		err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+			found, err := lookForStringInPodExec(corednspod.Namespace, corednspod.Name, "dns", catCmd, "class denial error")
+			if err != nil {
+				t.Logf("%s/%s: cat error: %v", corednspod.Namespace, corednspod.Name, err)
+				return false, err
 			}
-			cliImage = ver.Version
-			cliImageFound = true
-			break
-		}
-	}
-	if !cliImageFound {
-		t.Fatalf("failed to find the %s version for clusteroperator %s", statuscontroller.OpenshiftCLIVersionName, co.Name)
-	}
-
-	// Create the client Pod.
-	testClientForDNSLogging := buildPod("test-client-dnslogging", upstreamPodNs, cliImage, []string{"sleep", "3600"})
-	if err := cl.Create(context.TODO(), testClientForDNSLogging); err != nil {
-		t.Fatalf("failed to create pod %s/%s: %v", testClientForDNSLogging.Namespace, testClientForDNSLogging.Name, err)
-	}
-	defer func() {
-		if err := cl.Delete(context.TODO(), testClientForDNSLogging); err != nil {
-			t.Fatalf("failed to delete pod %s/%s: %v", testClientForDNSLogging.Namespace, testClientForDNSLogging.Name, err)
-		}
-	}()
-	// Wait for the client Pod to be ready.
-	name := types.NamespacedName{Namespace: testClientForDNSLogging.Namespace, Name: testClientForDNSLogging.Name}
-	err = wait.PollImmediate(1*time.Second, 60*time.Second, func() (bool, error) {
-		if err := cl.Get(context.TODO(), name, testClientForDNSLogging); err != nil {
-			t.Logf("failed to get pod %s/%s: %v", name.Namespace, name.Name, err)
-			return false, nil
-		}
-		for _, cond := range testClientForDNSLogging.Status.Conditions {
-			if cond.Type == corev1.ContainersReady &&
-				cond.Status == corev1.ConditionTrue {
-				return true, nil
+			if !found {
+				t.Logf("%s/%s: failed to match 'class denial error' in %q, retrying...", corednspod.Namespace, corednspod.Name, catCmd[1])
+				return false, nil
 			}
-		}
-		return false, nil
-	})
-	if err != nil {
-		t.Fatalf("failed to observe ContainersReady condition for pod %s/%s: %v", testClientForDNSLogging.Namespace, testClientForDNSLogging.Name, err)
-	}
 
-	found := 0
-	for _, corednspod := range coreDNSPods.Items {
-
-		// Dig the example dns forwarding host.
-		digCmd := []string{"dig", "test.svc.cluster.local"}
-		if err := lookForStringInPodExec(testClientForDNSLogging.Namespace, testClientForDNSLogging.Name, testClientForDNSLogging.Name, digCmd, "NXDOMAIN", 2*time.Minute); err != nil {
-			t.Fatalf("failed to dig %v", err)
-		}
-
-		if err := lookForStringInPodExec(corednspod.Namespace, corednspod.Name, "dns", catCmd, "class denial error", 5*time.Minute); err != nil {
-			t.Fatalf("failed to set Debug logLevel for operator %s: %v", opName, err)
-		}
+			host := fmt.Sprintf("test-%s-%v.svc.cluster.local", corednspod.Name, time.Now().UnixMicro())
```

Comment:
Related to my above comment, maybe it's best to drop some more block comments about what we are doing:
```suggestion
                       // Run a dig query that locally queries this DNS pod (localhost) to trigger a log entry in the CoreDNS pod.
			host := fmt.Sprintf("test-%s-%v.svc.cluster.local", corednspod.Name, time.Now().UnixMicro())
```